packages/app/app/components/FavoriteTracksView/index.tsx

```diff
@@ -34,6 +41,25 @@ const FavoriteTracksView: React.FC<FavoriteTracksViewProps> = ({
   removeFavoriteTrack
 }) => {
   const { t } = useTranslation('favorites');
+  const dispatch = useDispatch();
+  const settings = useSelector(settingsSelector);
+
+  const toggleOption = useCallback(
+    (option, state) => dispatch(settingsActions.toggleOption(option, state)), [dispatch]
+  );
+
+  const toggleShuffle = useToggleOptionCallback(toggleOption, 'shuffleQueue', settings);
```

Comment:
Is this the correct way to use `useToggleOptionCallback`? I tried to extract the logic to specifically only toggle shuffle queue but it didn't seem to work 🤔

Reply:
That looks good to me